src/coreclr/src/jit/gentree.cpp

```diff
@@ -3483,7 +3483,10 @@ unsigned Compiler::gtSetEvalOrder(GenTree* tree)
                 break;
 
             case GT_CNS_DBL:
-                level = 0;
+            {
+                var_types targetType = tree->TypeGet();
+                level                = 0;
+#if defined(TARGET_XARCH)
                 /* We use fldz and fld1 to load 0.0 and 1.0, but all other  */
```

Comment:
Unrelated to your change, but this isn't accurate anymore. `fldz` and `fld1` are from the x87 stack and we almost exclusively use SSE/SSE2 now (modulo some ABI things for 32-bit).

Now its that we use `xorps` for `0` and load values for everything else.